Tests/Core/AWSClientRuntimeTests/AWSClientConfigurationTests.swift

```diff
@@ -74,17 +75,40 @@ class AWSClientConfigurationTests: XCTestCase {
 }
 
 struct TestAWSServiceSpecificConfiguration: AWSServiceSpecificConfiguration {
+    var authSchemes: Array<ClientRuntime.AuthScheme>
+    
+    var authSchemeResolver: ClientRuntime.AuthSchemeResolver
+    
     struct EndpointResolver {}
 
     typealias AWSServiceEndpointResolver = EndpointResolver
 
     var endpointResolver: EndpointResolver
 
-    init(endpointResolver: EndpointResolver?) throws {
+    init(endpointResolver: EndpointResolver?,
+         authSchemeResolver: ClientRuntime.AuthSchemeResolver?,
+         authSchemes: [ClientRuntime.AuthScheme]?
+    ) throws {
         self.endpointResolver = endpointResolver ?? EndpointResolver()
+        self.authSchemes = authSchemes ?? []
```

Comment:
Question: Will we need to do anything new/special to support optionalAuth trait?
https://smithy.io/2.0/spec/authentication-traits.html#optionalauth-trait

Nit: If `[]` is a reference to supporting DisabledAuth maybe we can add a comment with a link to what this means

Reply:
SRA Identity & Auth work will handle optionalAuth trait as part of it (in fact the codegen for service specific `AuthSchemeResolver`s that's added in the [smithy-swift PR](https://github.com/smithy-lang/smithy-swift/pull/609) currently handles optoinal auth, there's nothing new/special that's needed. 

Also - `[]` isn't a reference to anything, it was just the simplest filler that compiler stopped complaining about - no additional tests have been added in this file that tests auth schemes.